crates/egui/src/layers.rs

```diff
@@ -149,6 +149,10 @@ impl PaintList {
     /// and then later setting it using `paint_list.set(idx, cr, frame);`.
     #[inline(always)]
     pub fn set(&mut self, idx: ShapeIdx, clip_rect: Rect, shape: Shape) {
+        if self.0.len() <= idx.0 {
+            self.add(clip_rect, Shape::Noop);
+        }
```

Comment:
This shouldn't be needed unless the user does something wrong, and even then this only works if it is exactly off-by-one

Reply:
> This shouldn't be needed unless the user does something wrong, and even then this only works if it is exactly off-by-one

I experienced a panic occurring when self.0.len() was 0, and there is no problem if I use this. glow_integration and wgpu_integration must also be changed. I will add that part to this commit as well.

How to reproduce panic:
1. Run examples/multiple_windows.
2. Open all viewports.
3. Close “DD -> Top D -> SS -> Top S”. (order required)

Reply:
I fail to reproduce.

In any case: try to find the root of the problem instead of adding defensive solutions

Reply:
And use `RUST_BACKTRACE=1` to get stack traces